src/main/java/network/brightspots/rcv/HartCvrReader.java

```diff
@@ -0,0 +1,169 @@
+  private final ContestConfig contestConfig;
+
+  HartCvrReader(String cvrPath, ContestConfig contestConfig) {
+    this.cvrPath = cvrPath;
+    this.contestConfig = contestConfig;
+  }
+
+  // iterate all xml files in the source input folder
+  void readCastVoteRecordsFromFolder(List<CastVoteRecord> castVoteRecords)
+      throws IOException, CvrParseException {
+    File cvrRoot = new File(this.cvrPath);
+    File[] children = cvrRoot.listFiles();
+    for (File child : children) {
+      if (child.getName().endsWith("xml")) {
+        readCastVoteRecord(castVoteRecords, child.toPath());
+      }
+    }
+  }
+
+  // parse Cvr xml file into CastVoteRecord objects and add them to the input List<CastVoteRecord>
+  void readCastVoteRecord(List<CastVoteRecord> castVoteRecords, Path path)
+      throws IOException, CvrParseException {
+    try {
+      Logger.log(Level.INFO, "Reading Hart cast vote record file: %s...", path.getFileName());
+
+      XmlMapper xmlMapper = new XmlMapper();
+      xmlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      String xmlString = new String(Files.readAllBytes(path));
+      HartCvrXml xmlCvr = xmlMapper.readValue(xmlString, HartCvrXml.class);
+
+      for (Contest contest : xmlCvr.Contests) {
+        // TODO: use contest Id
+        if (!contest.Name.equals(contestConfig.getContestName())) {
+          continue;
+        }
+        if (contest.Options == null) {
+          // undervoted
+          continue;
+        }
+
+        ArrayList<Pair<Integer, String>> rankings = new ArrayList<>();
+        for (Option option : contest.Options) {
+          if (option.WriteInData != null) {
+            Logger.log(Level.WARNING, "");
+          }
+
+          if (!this.contestConfig.getCandidateCodeList().contains(option.Id)) {
+            Logger.log(
+                Level.SEVERE,
+                "Candidate ID: \"%s\" name: \"%s\" from CVR is not in the config file!",
+                option.Id, option.Name);
+            throw new CvrParseException();
+          }
+
+          // Hart RCV election ranks are indicated by a string read left to right:
+          // each digit corresponds to a rank and is set to 1 if that rank was voted.  For example:
+          // 0101 indicates ranks 2 and 4 are voted for this option (overvote)
+          // 00   indicates no rank was voted
+          // 010  indicates rank 2 was voted
+          for (int rank = 1; rank < option.Value.length() + 1; rank++) {
+            String rankValue = option.Value.substring(rank - 1, rank);
+            if (rankValue.equals("1")) {
+              rankings.add(new Pair<>(rank, option.Id));
+            }
+          }
+        }
+
+        CastVoteRecord cvr = new CastVoteRecord(
+            contest.Id,
+            null,
+            xmlCvr.BatchNumber,
+            xmlCvr.CvrGuid,
+            xmlCvr.PrecinctSplit.Name,
+            xmlCvr.PrecinctSplit.Id,
+            null,
+            rankings);
```

Comment:
remember to run the style thing before you commit to develop

Reply:
(but don't run it on your JSON files)

Reply:
Yeah, I actually remembered this time.  Were you expecting different formatting here?  This is what I get from intelliJ 'reformat code'.  Google checkstyle does not care about this either.

Reply:
Oh, I thought it would put the close parens on its own line, since that's what I always do and it doesn't change it. I guess it doesn't enforce a consistent style for this...